lib/model/CDataGatherer.cc

```diff
@@ -820,16 +819,14 @@ bool CDataGatherer::restoreBucketGatherer(const std::string& summaryCountFieldNa
                 *this, summaryCountFieldName, personFieldName, attributeFieldName,
                 valueFieldName, influenceFieldNames, traverser);
             if (m_BucketGatherer == nullptr) {
-                LOG_ERROR(<< "Failed to create event rate bucket gatherer");
-                return false;
+                LOG_ABORT(<< "Failed to create event rate bucket gatherer");
             }
         } else if (name == CBucketGatherer::METRIC_BUCKET_GATHERER_TAG) {
             m_BucketGatherer = std::make_unique<CMetricBucketGatherer>(
                 *this, summaryCountFieldName, personFieldName, attributeFieldName,
                 valueFieldName, influenceFieldNames, traverser);
             if (m_BucketGatherer == nullptr) {
-                LOG_ERROR(<< "Failed to create metric bucket gatherer");
-                return false;
+                LOG_ABORT(<< "Failed to create metric bucket gatherer");
```

Comment:
I'm not convinced that this is correct. `std::make_unique...` isn't normally going to be returning null and we're missing the case that the state doesn't have a `EVENTRATE_BUCKET_GATHERER_TAG` or a `METRIC_BUCKET_GATHERER_TAG`. I'd just have a check that it's non-null after the loop.